server/graphql/v1/queries.js

```diff
@@ -874,9 +874,17 @@ const queries = {
       id: { type: GraphQLInt },
       code: { type: GraphQLString },
     },
-    resolve(_, args) {
+    async resolve(_, args, req) {
       if (args.id) {
-        return models.PaymentMethod.findByPk(args.id);
+        const paymentMethod = await models.PaymentMethod.findByPk(args.id, {
+          include: [{ model: models.Collective, required: true }],
+        });
+
+        if (!paymentMethod || !req.remoteUser?.isAdminOfCollective(paymentMethod.Collective)) {
```

Comment:
This permission check wasn't really an issue (we already check private fields) but it still allowed to enumerate all payment methods. We only fetch by `id` on the page to update the payment method, so there should be no impact.